### ISSUE TITLE
Showing a formspec shortly after another one was closed works rarely

### ISSUE THREAD

**Issue by appgurueu** [minetest/minetest#11907]
##### Minetest version
```
5.4.1
```

##### Summary
https://github.com/minetest/minetest/issues/11898 apparently isn't limited to reshowing formspecs. Showing *any* formspec right after another one was closed (using a `button_exit[]` for instance). The simple `minetest.after(0, ...)` workaround for delaying the showing of the second formspec doesn't seem to always work either; the larger the waiting time, the higher the chance of the second formspec showing up. Converting exit buttons to simple buttons - not closing the formspec, replacing it instead - seems to work though.

##### How to reproduce

```lua
minetest.register_on_joinplayer(function(player)
	minetest.show_formspec(player:get_player_name(), "fs_bug_test:stage_one", "field[proceed;Stage one!;close this]")
end)

minetest.register_on_player_receive_fields(function(player, fields)
	minetest.show_formspec(player:get_player_name(), "fs_bug_test:stage_two", "field[proceed2;Stage two!;yay]")
end)
```

You should be shown a formspec. Close it. If no second formspec opens, there is a bug.

**Comment by v-rob** [comment]
I isolated the problem:

https://github.com/minetest/minetest/blob/8fab406c28bc5a18137d2653356c10880c827613/src/gui/guiFormSpecMenu.cpp#L4571-L4578

As far as I can tell, on a button_exit[] being clicked, acceptInput() is called, which sends the fields to the server.  The client then intends to call quitMenu().  The server receives the fields, sends it to Lua, and promptly sends this new formspec to the client again, all _before_ quitMenu() is called.  When the client then calls quitMenu(), it destroys the new formspec as well.  In other words, a race condition between client and server.  Commenting out quitMenu() allows the new formspec to be opened, but of course doesn't allow formspecs to close normally.

I've confirmed this bug as far back as 5.0.0, but it's probably very, very old.
